experimental/zerotrie/src/varint.rs

```diff
@@ -4,17 +4,48 @@
 
 //! Varint spec for ZeroTrie:
 //!
-//! - Lead byte: top 2 bits are trie metadata; third is varint extender; rest is value
-//! - Trail bytes: top bit is varint extender; add rest to current value * 2^7
-//! - Add the "latent value" to the final result: (1<<5) + (1<<7) + (1<<14) + ...
+//! - Lead byte: top M (2 or 3) bits are metadata; next is varint extender; rest is value
+//! - Trail bytes: top bit is varint extender; rest are low bits of value
+//! - Guaranteed uniqueness of varint by adding "latent value" for each extender byte
+//! - No maximum, but high bits will be dropped if they don't fit in the platform's `usize`
+//!
+//! This is best shown by examples.
+//!
+//! ```txt
+//! xxx0'1010 = 10
+//! xxx0'1111 = 15 (largest single-byte value with M=3)
+//! xxx1'0000 0000'0000 = 16 (smallest two-byte value with M=3)
```

Comment:
the `=` signs are a bit confusing here, i think it's worth splitting this as "_if_ the largest single byte valueis 15, then the smallest two-byte value must be 16", because these examples are both establishing the concept of the latent value and also using it

Reply:
maybe just replacing `=` with "must be"

Reply:
Done, for the two rows that introduce new latent values

Reply:
Done, for the two rows that introduce new latent values

Reply:
Done, for the two rows that introduce new latent values